### PR TITLE
add client-side middleware in worker

### PR DESCRIPTION
**Pull request by ruban-thilak**
Adding a client_middleware array on Lowkiq module. It can be useful to add client-side middlewares (i.e: Loggers, Opentelemetry tracer, etc.) 

We construct a Lamda method with those middlewares present in the client_middleware array. (Please check client_wrapper method in Lowkiq.rb)

The client_wrapper wraps around the perform_async method in Lowkiq::Worker module and executes all the middlewares everytime a job is enqueued with perform_async method.

This implementation is similar to the existing server_wrapper method in Lowkiq.rb.

### CODE REVIEW

File: lib/lowkiq.rb [hunk 1]
```diff
@@ -64,9 +64,13 @@ def server_wrapper
...
     end
 
+    def client_wrapper
+      @client_middleware ||= self.middleware_wrapper(self.client_middlewares)
+    end
+
     def shard_handlers
       self.workers.flat_map do |w|
-        ShardHandler.build_many w, self.server_wrapper
+        ShardHandler.build_many w, self.middleware_wrapper(self.server_middlewares)
```
Comment (darkleaf) [inline lib/lowkiq.rb]: I think you should keep `self.server_wrapper` and implement it like `client_wrapper`.

Comment (ruban-thilak) [reply lib/lowkiq.rb]: Refactored the code as suggested.

File: lib/lowkiq.rb [hunk 2]
```diff
@@ -53,9 +53,9 @@ def client_redis_pool
       @client_redis_pool ||= ConnectionPool.new(size: client_pool_size, timeout: pool_timeout, &redis)
     end
 
-    def middleware_wrapper(middlewares)
+    def server_wrapper
```
Comment (darkleaf) [inline lib/lowkiq.rb]: You misunderstand me.

You should keep `middleware_wrapper`, `client_wrapper` and add `server_wrapper` like `client_wrapper`.

```ruby
def client_wrapper
      @client_wrapper ||= self.middleware_wrapper(self.client_middlewares)
end
```

and also add 

```ruby
def server_wrapper
      @server_wrapper ||= self.middleware_wrapper(self.server_middlewares)
end
```

Comment (ruban-thilak) [reply lib/lowkiq.rb]: Sorry, I misunderstood. I have corrected it. You can review it now.

**Comment by darkleaf** [comment]
@ruban-thilak nice.
Please update Readme#Configuration.

**Comment by ruban-thilak** [comment]
@darkleaf done

**Comment by darkleaf** [comment]
1.2.2 🥳 
cc: @solutus

**Pull request merged**